metals/src/main/scala/scala/meta/internal/builds/BuildTools.scala

```diff
@@ -121,21 +121,40 @@ final class BuildTools(
   )
   def isBazel: Boolean = bazelProject.isDefined
 
+  private def customProjectRoot =
+    userConfig().customProjectRoot
+      .map(_.trim() match {
+        case "." => workspace
+        case relativePath => workspace.resolve(relativePath)
+      })
+      .filter { projectRoot =>
+        val exists = projectRoot.exists
+        if (!exists) {
+          scribe.error(s"custom project root $projectRoot does not exist")
+        }
+        exists
+      }
+
   private def searchForBuildTool(
       isProjectRoot: AbsolutePath => Boolean
-  ): Option[AbsolutePath] =
-    if (isProjectRoot(workspace)) Some(workspace)
-    else
-      workspace.toNIO
-        .toFile()
-        .listFiles()
-        .collectFirst {
-          case file
-              if file.isDirectory &&
-                !file.getName.startsWith(".") &&
-                isProjectRoot(AbsolutePath(file.toPath())) =>
-            AbsolutePath(file.toPath())
-        }
+  ): Option[AbsolutePath] = {
+    customProjectRoot match {
+      case Some(projectRoot) => Some(projectRoot).filter(isProjectRoot)
```

Comment:
```suggestion
      case Some(projectRoot) if isProjectRoot(projectRoot) => Some(projectRoot)
```

Reply:
It's on purpose like this. So if you choose a custom root it should look only in that root. We had a problem discussed on discord, where a person had a custom bsp and an unused `sbt`  somewhere lower file hierarchy.